src/platform/datapath_raw_socket.c

```diff
@@ -342,12 +342,77 @@ CxPlatDpRawCalculateHeaderBackFill(
     return HeaderBackFill;
 }
 
+_IRQL_requires_max_(DISPATCH_LEVEL)
+uint16_t
+CxPlatFramingChecksum(
+    _In_reads_(Length) uint8_t* Data,
+    _In_ uint32_t Length,
+    _In_ uint64_t InitialChecksum
+    )
+{
+    //
+    // Checksum is calculated in reverse order.
```

Comment:
I assumed the first byte of the buffer passed down to XDP would be looked at first. I didn't know CPUs have logic to prefetch the next cache line. So, forwards direction might make more sense.

Reply:
Yeah. CPUs also have logic to prefetch based on what will likely happen next as the code executes, so it might be pointless. Still, it's also just more common to iterate through an array forwards.